pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -53,21 +56,95 @@ const (
 // DefaultPreemption is a PostFilter plugin implements the preemption logic.
 type DefaultPreemption struct {
 	fh        framework.FrameworkHandle
+	args      config.DefaultPreemptionArgs
 	podLister corelisters.PodLister
 	pdbLister policylisters.PodDisruptionBudgetLister
+	// candidateOffset maintains state about the last offset used for dry running
+	// preemption.
+	candidateOffset int32
 }
 
 var _ framework.PostFilterPlugin = &DefaultPreemption{}
 
+type FindCandidatesOptions struct {
+	CandidateOffset             *int32
+	MinCandidateNodesPercentage int32
+	MinCandidateNodesAbsolute   int32
+}
+
+type FindCandidatesOption func(*FindCandidatesOptions)
```

Comment:
I honestly feel like we are using a big hammer to hit a small nail :) I don't think the "With option" pattern applicable to what we need here, too much boiler plate for just calculating the number of nodes and offset.

Reply:
For in-tree defaultpreemption plugin, it does. However, it's more for out-of-tree plugins to use FindCandidates() while also enjoying the feature of looping over fewer candidates.